main.go

```diff
@@ -156,6 +160,12 @@ func main() {
 		os.Exit(1)
 	}
 
+	if scopeCacheMaxSize <= 0 {
+		err := errors.New("--scope-cache-max-size should be > 0")
+		setupLog.Error(err, "unable to start manager")
+		os.Exit(1)
+	}
```

Comment:
What do you think about allowing 0 cache size as a way to disable caching?
I'm not sure if that is something we need to support but it is often beneficial to be able to opt-out of new features

Reply:
Good idea! 
Another benefit we can have is to debug without cache if something is going wrong

Reply:
Updated, setting the flag to `--scope-cache-max-size=0` disable the cache.